internal/docker/driver.go

```diff
@@ -32,47 +38,66 @@ func (d *dockerDriver) Type() string {
 }
 
 // StartProcess helps implement dockerDriver
-func (d *dockerDriver) StartProcess(process *node.Process, app *node.AppInstallation) (string, error) {
+func (d *dockerDriver) StartProcess(ctx context.Context, process *node.Process, app *node.AppInstallation) error {
 	var dockerConfig node.AppInstallationConfig
 	dockerConfigBytes, err := json.Marshal(app.DriverConfig)
 	if err != nil {
-		return "", err
+		return err
 	}
 
 	err = json.Unmarshal(dockerConfigBytes, &dockerConfig)
 	if err != nil {
-		return "", err
+		return err
 	}
 
 	exposedPorts := make(nat.PortSet)
 	for _, port := range dockerConfig.ExposedPorts {
 		exposedPorts[nat.Port(port)] = struct{}{}
 	}
 
-	createResp, err := d.client.ContainerCreate(context.Background(), &container.Config{
+	createResp, err := d.client.ContainerCreate(ctx, &container.Config{
 		Image:        fmt.Sprintf("%s/%s:%s", app.RegistryURLBase, app.RegistryPackageID, app.RegistryPackageTag),
 		ExposedPorts: exposedPorts,
 		Env:          dockerConfig.Env,
+		Labels: map[string]string{
+			habitatLabel: string(process.ID),
+		},
 	}, &container.HostConfig{
 		PortBindings: dockerConfig.PortBindings,
 		Mounts:       dockerConfig.Mounts,
 	}, nil, nil, "")
 	if err != nil {
-		return "", err
+		return err
 	}
 
-	err = d.client.ContainerStart(context.Background(), createResp.ID, container.StartOptions{})
+	err = d.client.ContainerStart(ctx, createResp.ID, container.StartOptions{})
 	if err != nil {
-		return "", err
+		return err
 	}
 
-	log.Info().Msgf("Started container %s", createResp.ID)
+	log.Info().Msgf("Started docker container %s", createResp.ID)
 
-	return createResp.ID, nil
+	return nil
 }
 
-func (d *dockerDriver) StopProcess(extProcessID string) error {
-	err := d.client.ContainerStop(context.Background(), extProcessID, container.StopOptions{})
+func (d *dockerDriver) StopProcess(ctx context.Context, processID node.ProcessID) error {
+	ctrs, err := d.client.ContainerList(ctx, container.ListOptions{})
```

Comment:
The `ListOptions` struct probably has a way to filter by labels. It probably makes more sense to use docker's implementation here.